src/mam4xx/aero_model_emissions.hpp

```diff
@@ -0,0 +1,919 @@
+// FIXME: fortran index?
+const int organic_num_idx[organic_num_modes] = {0, 1, 3};
+// =============================================================================
+
+KOKKOS_INLINE_FUNCTION
+void init_dust_dmt_vwr(int nbin, const Real (&dust_dmt_grd)[dust_nbin + 1],
+                       Real (&dust_dmt_vwr)[dust_nbin]) {
+
+  // FIXME: BAD CONSTANT
+  const int sz_nbr = 200;
+
+  // Introducing particle diameter. Needed by atm model and by dry dep model.
+  // Taken from Charlie Zender's subroutines dst_psd_ini, dst_sz_rsl,
+  // grd_mk (dstpsd.F90) and subroutine lgn_evl (psdlgn.F90)
+
+  // Charlie allows logarithmic or linear option for size distribution.
+  // however, he hardwires the distribution to logarithmic in his code.
+  // therefore, I take his logarithmic code only
+  // furthermore, if dst_nbr == 4, he overrides the automatic grid calculation
+  // he currently works with dst_nbr = 4, so I only take the relevant code
+  // if dust_number ever becomes different from 4,
+  // must add call grd_mk (dstpsd.F90) as done in subroutine dst_psd_ini
+  // note that here dust_number = dst_nbr
+
+  // Max diameter in bin [m]
+  Real dmt_min[nbin];
+  // Min diameter in bin [m]
+  Real dmt_max[nbin];
+
+  // Override automatic grid with preset grid if available
+  for (int n = 0; n < nbin; ++n) {
+    dmt_min[n] = dust_dmt_grd[n];
+    dmt_max[n] = dust_dmt_grd[n + 1];
+  }
+
+  // set dust_dmt_vwr ....
+
+  // Bin physical properties
+  // [frc] Geometric std dev PaG77 p. 2080 Table1
+  Real gsd_anl = 2.0;
+  Real ln_gsd = haero::log(gsd_anl);
+
+  // Set a fundamental statistic for each bin
+  // Mass median diameter analytic She84 p.75 Table 1 [m]
+  // FIXME: FIXME: this appears to be a bug--vma is set, and then re-set
+  //        maybe the second one should be dmt_nma?
+  Real dmt_vma = 2.524e-6;
+  dmt_vma = 3.5e-6;
+  // Compute analytic size statistics
+  // Convert mass median diameter to number median particle diameter [m]
+  Real dmt_nma = dmt_vma * haero::exp(-3.0 * haero::square(ln_gsd));
+
+  // [m] Size Bin minima
+  Real sz_min[sz_nbr];
+  // [m] Size Bin maxima
+  Real sz_max[sz_nbr];
+  // [m] Size Bin centers
+  Real sz_ctr[sz_nbr];
+  // [m] Size Bin widths
+  Real sz_dlt[sz_nbr];
+
+  // [m3 m-3] Volume concentration resolved
+  Real vlm_rsl[nbin] = {0.0};
+
+  // Compute resolved size statistics for each size distribution
+  // In C. Zender's code call dst_sz_rsl
+  // Factor for logarithmic grid
+  Real series_ratio;
+  for (int n = 0; n < nbin; ++n) {
+    series_ratio = haero::pow((dmt_max[n] / dmt_min[n]), (1.0 / sz_nbr));
+    sz_min[1] = dmt_min[n];
+    // NOTE: Loop starts at 1 (2 in fortran code)
+    for (int m = 1; m < sz_nbr; ++m) {
+      sz_min[m] = sz_min[m - 1] * series_ratio;
+    }
+    // Derived grid values
+    // NOTE: Loop ends at sz_nbr - 2 (fortran: sz_nbr - 1)
+    for (int m = 0; m < sz_nbr - 2; ++m) {
+      sz_max[m] = sz_min[m + 1];
+    }
+    sz_max[sz_nbr] = dmt_max[n];
```

Comment:
Index out-of-bounds: `sz_max[sz_nbr] `should be `sz_max[sz_nbr-1]`.

Reply:
Clearly transcribing 1-based to 0-based languages is my kryptonite... 🙃

If there's one thing I can hope to accomplish in my career, it'd be to develop those **_J. Overfelt Eagle Eyes_** ™️ 🙂

Thank you!